src/main/java/frc/robot/subsystems/drivetrain/MissileShellSwerveSubsystem.java

```diff
@@ -35,6 +41,32 @@ public MissileShellSwerveSubsystem() {
     public void periodic() {
         SwerveDriveKinematics.desaturateWheelSpeeds(states, MAX_VEL);
         module.setDesiredState(states[0]);
+
+        // Print values for encoder tuning. To set a zero, offset + currentAngle = 0 -> offset = -currentAngle.
+        // In case the offset is 180 degrees off, also print it plus pi.
+        if (OFFSET_TUNING_ENABLE) {
+            double currentAngleRads = module.getRawAngleRads();
+
+            // The robot offset is the offset of the module assuming that it is aligned with the front of the robot.
+            // For the robot offset, subtract the position offset automatically applied by the module subclasses.
+            // currentAngle + offset + posOffset = 0 -> offset = -currentAngle - posOffset.
+            double robotOffset = MathUtil.angleModulus(-currentAngleRads - module.getPositionOffsetRads());
+            double flippedRobotOffset = MathUtil.angleModulus(-currentAngleRads - module.getPositionOffsetRads() + Math.PI);
+
+            // The pin offset is the offset of the module assuming that it is aligned with the aligning pin.
+            // For the pin offset, subtract pi/2 to account for the pin being 90 degrees off.
+            // currentAngle + offset + pi/2 = 0 -> offset = -currentAngle - pi/2.
```

Comment:
this seems like a potential source of confusion--why don't we assume that the pin is the 'correct' 0, instead of 90deg off?

Reply:
Because our convention had been that the top left module should face the front of the robot at 0 degrees, but the pin makes it face 90. I thought that it'd just be easier to subtract pi/2 in desmos than to change all of our previous non-pin offsets.